plugins/weather.rb

```diff
@@ -31,22 +31,23 @@ def match?(cmd)
 
   def weather(m, location)
     if KEY
-      begin
-        client = Apixu::Client.new KEY
-        hash = client.current location
-      rescue Apixu::Errors::Request => e
-        if e.code == Apixu::Errors::NO_LOCATION_FOUND_FOR_QUERY
-          m.reply "Nothing found for location \'#{location}\'.", true
-        else
-          m.reply e, true
-        end
-      else
+        params = {
+          :access_key => KEY,
+          :query => location
+        }
+        uri = URI("http://api.weatherstack.com/current")
+        uri.query = URI.encode_www_form(params)
+        json = Net::HTTP.get(uri)
+        hash = JSON.parse(json)
+      if hash["location"]
         loc = hash["location"]["name"]
-        loc = "#{loc}, #{hash["location"]["region"]}" unless hash["location"]["region"].empty?
+        loc = "#{loc}, #{hash["location"]["region"]}"
         loc = "#{loc}, #{hash["location"]["country"]}"
-        weather = hash["current"]["condition"]["text"]
-        temp = "#{hash["current"]["temp_c"]}°C (#{hash["current"]["temp_f"]}°F)"
+        weather = hash["current"]["weather_descriptions"][0]
+        temp = "#{hash["current"]["temperature"]}°C (#{hash["current"]["temperature"]*1.8+32}°F)"
```

Comment:
Let's break the fahrenheit calculation out into another line, instead of doing it in the interpolation.

Reply:
so that would be a separate `temp = "#{temp} (#{hash["current"]["temperature"]*1.8+32}°F)"` right after
or 
```
temp_f = hash["current"]["temperature"]*1.8+32
temp = "#{hash["current"]["temperature"]}°C (#{temp_f}°F)" 
```
?

Reply:
possibly just do temp_c and temp_f separately and combine in the end

Reply:
Yeah, `temp_c` and `temp_f` sounds reasonable to me.